app/gui/src/model/suggestion_database/entry.rs

```diff
@@ -54,6 +54,61 @@ pub struct MissingThisOnMethod(pub String);
 
 
 
+// =====================
+// === QualifiedName ===
+// =====================
+
+im_string_newtype! {
+    /// A single segment of a [`QualifiedName`] of an [`Entry`].
+    QualifiedNameSegment
+}
+
+/// A fully qualified name of an [`Entry`].
+#[derive(Debug, Default, Clone, PartialEq)]
+#[allow(missing_docs)]
+pub struct QualifiedName {
+    pub segments: Vec<QualifiedNameSegment>,
+}
+
+impl From<QualifiedName> for String {
+    fn from(name: QualifiedName) -> Self {
+        String::from(&name)
+    }
+}
+
+impl From<&QualifiedName> for String {
+    fn from(name: &QualifiedName) -> Self {
+        name.into_iter().map(|s| s.deref()).join(ast::opr::predefined::ACCESS)
+    }
+}
```

Comment:
Add these to `im_string_newtype` instead.

Reply:
AFAIU this suggestion is not applicable: the highlighted methods are defined on `QualifiedName`, whereas `im_string_newtype` is used to define `QualifiedNameSegment`.